src/plugins/ncr/components/labels-picker/index.jsx

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import chroma from 'chroma-js';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import isEqual from 'lodash/isEqual';
+import labelColors from 'config/labels'; // eslint-disable-line import/no-unresolved
+import Message from '@gdbots/pbj/Message';
+import createDelegateFactory from '@triniti/app/createDelegateFactory';
+import {
+  Button,
+  Card,
+  CardBody,
+  Col,
+  Icon,
+  Row,
+} from '@triniti/admin-ui-plugin/components';
+import PicklistPicker from '@triniti/cms/plugins/sys/components/picklist-picker';
+import delegateFactory from './delegate';
+
+class Labels extends React.PureComponent {
+  static propTypes = {
+    disabled: PropTypes.bool,
+    disabledReasonMessage: PropTypes.string,
+    delegate: PropTypes.shape({
+      handleApplyLabels: PropTypes.func.isRequired,
+    }).isRequired,
+    node: PropTypes.instanceOf(Message),
+  };
+
+  static defaultProps = {
+    disabled: false,
+    disabledReasonMessage: 'Labels disabled until saved.',
+    node: null,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      touched: false,
+      selected: [],
+    };
+
+    this.isNodeDiff = this.isNodeDiff.bind(this);
+    this.handleApplyLabels = this.handleApplyLabels.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  UNSAFE_componentWillReceiveProps({ node }) {
+    if (!node) {
+      return;
+    }
+    const { touched } = this.state;
+    if (touched) {
+      return;
+    }
+    this.setState({
+      selected: node.get('labels', []),
+    });
+  }
+
+  handleApplyLabels() {
+    const { delegate } = this.props;
+    const { selected } = this.state;
+    delegate.handleApplyLabels(selected);
+  }
+
+  handleChange(selected) {
```

Comment:
nit: is it the same if you just map here? then you don't need the extra array.

` this.setState({
      selected: (selected || []).map(val => val.value),
      touched: true,
    });`